### ISSUE TITLE
Grouped images: always show grouping icon on image

### ISSUE THREAD

**Issue by inducer** [darktable-org/darktable#10688]
I just spent a fair while trying to figure out how to follow the instructions in the [grouping docs](https://docs.darktable.org/usermanual/3.8/en/lighttable/digital-asset-management/grouping/):

> You can define which image is considered to be the group leader by clicking on the group icon of the desired image while that group is expanded.

What I was missing is that, in order to show the grouping icon on the image, it is necessary to enable "permanent picture infos" (sorry, translating from German here) in the "star" menu. I was very confused because the grouping icon, which the docs said should be there, was not in fact shown.

As usability issues go, I think it might be better if the grouping icon on an image were always shown if it is part of a group... IMO there's not much of a reason not to. In fact, I think the grouping feature isn't really usable without that icon.

x-ref: https://github.com/darktable-org/dtdocs/issues/406

**Comment by elstoc** [comment]
Ping @AlicVB

**Comment by AlicVB** [comment]
pong @Nilvus :) (yes, I shamefully dump the work on you... but it's css... and you are way better than me in that sort of UI details)

in fact, I guess that there can be a good reason for all the other icons too... but the default mode is designed to be as light as possible for a very general usage, and imho this case is quite specific... But at least we have to specify that point in the user manual !

**Comment by Nilvus** [comment]
If it's css,I will so check asap

**Comment by Nilvus** [comment]
Checking and thinking more about this, I indeed think that group icon should appear same way as stars as this is something actionable. If it's ok for all, I can change that.

But so, we need to think more about of consistency on those overlays. Actually, we have this consistency:

- all permanent overlays display all icons (ext name, group, altered, icon, stars and reject ones)
- extended means display text infos
- no overlays means no infos (and this should remains the same, so even group icon will never be displayed)

So remains, overlays 2 and 3, so overlays on mouse hover and extended overlays on mouse hover.
Actually, only bottom infos are displayed on those 2 overlays. As they are both mouse hover, I'm wondering if it will not be better to simply display all things on mouse hover and so those 2 overlays become same as permanent but with the only difference that it is on mouse hover as permanent are... permanent.

WIll probably be more consistent and easier to see useful infos. Actually it's not possible to have all infos displayed only on mouse hover as some are not displayed.

At @elstoc, @AlicVB (and @TurboGit), what do you think about those changes?

**Comment by AlicVB** [comment]
@Nilvus : that seems pretty logical indeed. All good for me !

**Comment by TurboGit** [comment]
I will always vote for consistency when possible and here it makes sense to me.

**Comment by Nilvus** [comment]
Tested and so find how we didn't display them on overlays 2 and 3: just display them when hovering with a portrait image and see that extension name and group icon are displayed hover the image and it's hard to read. See below an example:

![image](https://user-images.githubusercontent.com/45535283/147486959-c7bc1a2f-5f21-4dc8-a6a0-d58397a8037b.png)

So do we add them as suggested here or do we let this as it is actually?

@inducer: as you created this issue, your view is also wanted. Thanks!

**Comment by Nilvus** [comment]
A third possibility is to add a specific background just for hover display as we already use it at the bottom. See below on same image and same overlay:

![image](https://user-images.githubusercontent.com/45535283/147487103-c1538aaa-239c-4110-ad9d-a431206e5302.png)

So what's vote for:

1. change nothing and let as it is. And so just update manual to explain top icons (like group one) are displayed only on overlays 4 to 7
2. add them on overlay even if they are not completely visible on some images (I don't really like this idea)
3. Add a subtle background like on bottom but that would make a sort of "sandwich" thing (but just when hovering). Thinking about that, I think we already talk about something like that when doing those overlays and finish with 1st option here.

Or someone have a fourth option to propose?

**Comment by inducer** [comment]
3 (most preferable) / 2 / 1 (least preferable) would be my ranking.

**Comment by Nilvus** [comment]
Ok, found when we discussed about that and why we made actual choice. Good to read again before doing new changes. See issue  #5300 and this comment on PR to fix it: https://github.com/darktable-org/darktable/pull/5303#issuecomment-640089525

Finally, 3 will be hard to read and need a rework of Gtk code (just tested and impossible by now to add that background possibly, by having a thumb_top where icons are inside. Probably a big work. @AlicVB: what do you think about that?

I'm even not sure if it's a really good choice. Not sure the "sandwich" way will be good and it will probably be worse on square image as they take all thumbnail background.

So, I'm definitely not for option 2 just because of square images where this will be worse (just too hard to read and could be seen as something not finished).

Option 3: I fear that the "sandwich" effect will not be good (but maybe it will be ok) and it's probably a lot of work to rewrite thumbnails to have a "thumb_top" widget to set this background. So, it's all yours if we go that way @AlicVB, for first big step at least! I can finish CSS work after.

Option 1: it's probably the best, easiest and faster way to deal with all that. Just think that it's only for group icon we talk about here (maybe a little for audio one but do we really have a lot of people that use audio in darktable...).

**Comment by AlicVB** [comment]
@Nilvus : great that you find the original discussion ! So I definitively change my vote for 1. I don't really like the "sandwich" solution... technically it should be doable, but aesthetically ....

That said, if all of you prefer 3, I'll handle the gtk changes

**Comment by TurboGit** [comment]
@Nilvus : This is tagged 3.8.1 but looking at the last exchange that's probably more for 4.0, right?

**Comment by Nilvus** [comment]
@TurboGit: indeed, I changed it

**Comment by TurboGit** [comment]
@Nilvus : can you give a status update here? Is that only a documentation work at last?

**Comment by TurboGit** [comment]
I do agree with open 1 for what it's worth.

**Comment by Nilvus** [comment]
Yes, it's just doc. I'm not sur to be able to spent time on it.

**Comment by elstoc** [comment]
Resolved in dtdocs (https://github.com/darktable-org/dtdocs/commit/446cf37cf05056e3e0a063f9796c42da05e8ab5a). Just a reminder though, that the main reason I went to the effort of moving the documentation to markdown format was to encourage the developers of new features to submit their own user manual changes (because it's so much easier now). 

I don't mind putting the work in to keep the manual up-to-date but I don't understand/use a lot of the new 4.0 features, so some assistance would be appreciated. I understand that English is not everyone's first/favourite language but even badly written but complete descriptions of new features are a significant leg-up for me.
